Neo4j.Driver/Neo4j.Driver/Internal/Connector/SocketClient.cs

```diff
@@ -40,6 +40,9 @@ private static class ProtocolVersion
         private IBoltWriter _writer;
         private readonly BufferSettings _bufferSettings;
 
+        private int _closedMarker = 0;
+        private bool _isOpen = false;
```

Comment:
Could we remove this flag and only keep `closedMarker`?
I think `closedMarker` can be initialized with `-1` and we can set it to `0` where we currently do `_isOpen = true`.